### PR TITLE
Feature/replace mongodb gdev 1416

### PR DESCRIPTION
**Pull request by mephenor**
```
Replaced MongoDB with HashiCorp Vault using hvac

Co-authored-by: KerstenBreuer <kersten-breuer@outlook.com>
Co-authored-by: Moritz Hahn <Moritz.Hahn@uni-tuebingen.de>
```

### CODE REVIEW

File: ekss/api/deps.py [hunk 1]
```diff
@@ -15,9 +15,19 @@
...
 
-def get_config():
-    """Get runtime configuration."""
+
+def config_injector():
+    """Injectable config, overridable for tests"""
     return CONFIG
+
+
+def get_vault(config: VaultConfig) -> VaultAdapter:
```
Comment (KerstenBreuer) [inline ekss/api/deps.py:29]: You can chain dependencies:
```suggestion
def get_vault(config: VaultConfig = Depends(config_injector)) -> VaultAdapter:
```

File: ekss/api/download/router.py [hunk 2]
```diff
@@ -54,17 +49,17 @@ async def get_header_envelope(
     *,
     secret_id: str,
     client_pk: str,
-    dao: FileSecretDao = Depends(dao_injector),
+    vault_config: VaultConfig = Depends(config_injector)
```
Comment (KerstenBreuer) [inline ekss/api/download/router.py:52]:
```suggestion
    vault: VaultAdapter = Depends(get_vault)
```

File: ekss/api/upload/router.py [hunk 3]
```diff
@@ -57,11 +52,10 @@ async def dao_injector() -> FileSecretDao:
 async def post_encryption_secrets(
     *,
     envelope_query: models.InboundEnvelopeQuery,
-    dao: FileSecretDao = Depends(dao_injector),
+    vault_config: VaultConfig = Depends(config_injector),
```
Comment (KerstenBreuer) [inline ekss/api/upload/router.py:55]:
```suggestion
    vault: VaultAdapter = Depends(get_vault),
```

**Review by KerstenBreuer — DISMISSED** [review]

File: tests/integration/test_post_endpoint.py [hunk 4]
```diff
@@ -20,16 +20,16 @@
...
-from ekss.core.dao.mongo_db import FileSecretDao
 
-from ..fixtures.dao_keypair import dao_fixture  # noqa: F401
-from ..fixtures.dao_keypair import generate_keypair_fixture  # noqa: F401
-from ..fixtures.file_fixture import first_part_fixture  # noqa: F401
-from ..fixtures.file_fixture import FirstPartFixture
+from ..fixtures.file import first_part_fixture  # noqa: F401
+from ..fixtures.file import FirstPartFixture
+from ..fixtures.keypair import generate_keypair_fixture  # noqa: F401
+from ..fixtures.vault import vault_fixture  # noqa: F401
```
Comment (MoritzHahn1337) [inline tests/integration/test_post_endpoint.py:30]: Same as above

File: tests/unit/test_upload.py [hunk 5]
```diff
@@ -19,10 +19,10 @@
...
 from ekss.core.envelope_decryption import extract_envelope_content
 
-from ..fixtures.dao_keypair import dao_fixture  # noqa: F401
-from ..fixtures.dao_keypair import generate_keypair_fixture  # noqa: F401
-from ..fixtures.file_fixture import first_part_fixture  # noqa: F401
-from ..fixtures.file_fixture import FirstPartFixture
+from ..fixtures.file import first_part_fixture  # noqa: F401
+from ..fixtures.file import FirstPartFixture
+from ..fixtures.keypair import generate_keypair_fixture  # noqa: F401
+from ..fixtures.vault import vault_fixture  # noqa: F401
```
Comment (MoritzHahn1337) [inline tests/unit/test_upload.py:25]: Same as above

**Review by MoritzHahn1337 — DISMISSED** [review]

**Review by MoritzHahn1337 — APPROVED** [review]

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**